packages/typescript/src/browser/typescript-client-contribution.ts

```diff
@@ -65,7 +65,10 @@ export class TypeScriptClientContribution extends BaseLanguageClientContribution
     protected init(): void {
         this.preferences.onPreferenceChanged(e => {
             if (e.preferenceName === 'typescript.server.log') {
-                this.restart();
+                // avoid activation on startup if there is no other activation event
```

Comment:
IMHO it shouldn't fire on start. @akosyakov would you agree, this is an issue per se?

Reply:
We can avoid it for preference contributions i guess, but how would you avoid it for dynamically contributed preferences, i.e. from vscode extensions?

Reply:
btw there is `this.running` property

Reply:
thanks! :+1: updated